src/sdl/KeybindingsMenu.cpp

```diff
@@ -0,0 +1,275 @@
+
+    SDL_SetWindowResizable(this->window, false);
+}
+
+bool KeybindingsMenu::handleEvent(SDL_Event& event) {
+    if (!Window::handleEvent(event)) {
+        return false;
+    }
+
+    if (event.window.windowID == this->windowId) {        
+        // TODO manage events
+        switch(event.type) {
+            case SDL_EVENT_MOUSE_BUTTON_DOWN:
+                if (event.button.button == SDL_BUTTON_LEFT) {
+                    keybindingsMenuClicked = true;
+                }
+            break;
+            case SDL_EVENT_MOUSE_BUTTON_UP:
+                if (event.button.button == SDL_BUTTON_LEFT) {
+                    keybindingsMenuClicked = false;
+                }
+            break;
+            case SDL_EVENT_KEY_DOWN:
+                if (this->keyWaitingFor) {
+                    if (event.key.scancode >= SDL_SCANCODE_A && event.key.scancode <= SDL_SCANCODE_0) {
+                        if (event.key.scancode != SDL_SCANCODE_ESCAPE) {
+                            // The user pressed the new keybind for the chosen key
+                            for (auto& item : *keymap) {
+                                if (item.second == *this->keyWaitingFor) {
+                                    keymap->erase(item.first);
+                                    break;
+                                }
+                            }
+                            unsigned char pressed_key = *this->keyWaitingFor;
+                            keymap->insert_or_assign(static_cast<uint8_t>(event.key.scancode), pressed_key);
+                        }
+                        this->keyWaitingFor = nullptr;
+                    }
+                }
+            break;
+        }
+    }
+
+    return true;
+}
+
+void KeybindingsMenu::update() {
+    if (!this->mouseFocus) {
+        keybindingsMenuClicked = false;
+        return;
+    }
+}
+
+void KeybindingsMenu::render() {
+    float x,y;
+    SDL_GetMouseState(&x, &y);
+    Clay_SetLayoutDimensions((Clay_Dimensions) { (float) this->width, (float) this->height });
+    Clay_SetPointerState((Clay_Vector2) { x, y }, keybindingsMenuClicked);
+    Clay_UpdateScrollContainers(false, (Clay_Vector2) { 0, 0 }, 0.0166f);
+
+    Clay_SetLayoutDimensions((Clay_Dimensions) { (float) this->width, (float) this->height });
+    Clay_BeginLayout();
+    
+    CLAY({ .id = CLAY_ID("KeybindingsConfiguration"), .layout = { .sizing = { .width = CLAY_SIZING_GROW(0), .height = CLAY_SIZING_GROW(0) }, .padding = CLAY_PADDING_ALL(8), .childGap = 8, .layoutDirection = CLAY_TOP_TO_BOTTOM }, .backgroundColor = {250, 250, 250, 250} }) {
+        CLAY({ .layout = { .sizing = { .width = CLAY_SIZING_GROW(0) }, .childGap = 8} }) {
+            CLAY({ .layout = { .sizing = { .width = CLAY_SIZING_GROW(0) }, .padding = CLAY_PADDING_ALL(8), .childGap = 8, .childAlignment = { .y = CLAY_ALIGN_Y_CENTER } }, .backgroundColor = {90, 60, 40, 255} }) {
+                char keyboardCharacter = this->getKeyboardCharacter(0x1);
```

Comment:
For all of these please pass them as a string into toClayString otherwise there are issues when not hovering over the window

Reply:
<img width="214" height="289" alt="image" src="https://github.com/user-attachments/assets/f9fe68b5-c226-4c8a-b4c6-b9a3af22ed07" />

Reply:
Making another toClayString method that converts a char to a string then calls the original one is also fine